homeassistant/components/ovo_energy/sensor.py

```diff
@@ -0,0 +1,245 @@
+        """Update OVO Energy entity."""
+        now = datetime.utcnow()
+        try:
+            usage: OVODailyUsage = await self._client.get_daily_usage(
+                now.strftime("%Y-%m")
+            )
+            if usage is None or usage.gas is None:
+                _LOGGER.warning("No data found for %s", self._name)
+                self._available = False
+                return False
+            last_reading: OVODailyGas = usage.gas[len(usage.gas) - 1]
+            self._state = last_reading.consumption
+            self._attributes = {
+                "start_time": last_reading.interval.start,
+                "end_time": last_reading.interval.end,
+            }
+            self._available = True
+        except aiohttp.ClientError as exception:
+            _LOGGER.warning(exception)
+            self._available = False
+            return False
+        return True
+
+
+class OVOEnergyLastElectricityCost(OVOEnergySensor):
+    """Defines a OVO Energy last cost sensor."""
+
+    def __init__(self, client: OVOEnergy, currency: str):
+        """Initialize OVO Energy sensor."""
+        super().__init__(
+            client,
+            f"{DOMAIN}_{client.account_id}_last_electricity_cost",
+            "OVO Last Electricity Cost",
+            "mdi:cash-multiple",
+            currency,
+        )
+
+    async def _ovo_energy_update(self) -> bool:
+        """Update OVO Energy entity."""
+        now = datetime.utcnow()
+        try:
+            usage: OVODailyUsage = await self._client.get_daily_usage(
+                now.strftime("%Y-%m")
+            )
+            if usage is None or usage.electricity is None:
+                _LOGGER.warning("No data found for %s", self._name)
+                self._available = False
+                return False
+            last_reading: OVODailyElectricity = usage.electricity[
+                len(usage.electricity) - 1
+            ]
+            self._state = last_reading.cost.amount
+            self._attributes = {
+                "start_time": last_reading.interval.start,
+                "end_time": last_reading.interval.end,
+            }
+            self._available = True
+        except aiohttp.ClientError as exception:
+            _LOGGER.warning(exception)
+            self._available = False
+            return False
+        return True
+
+
+class OVOEnergyLastGasCost(OVOEnergySensor):
+    """Defines a OVO Energy last cost sensor."""
+
+    def __init__(self, client: OVOEnergy, currency: str):
+        """Initialize OVO Energy sensor."""
+        super().__init__(
+            client,
+            f"{DOMAIN}_{client.account_id}_last_gas_cost",
+            "OVO Last Gas Cost",
+            "mdi:cash-multiple",
+            currency,
+        )
+
+    async def _ovo_energy_update(self) -> bool:
+        """Update OVO Energy entity."""
+        now = datetime.utcnow()
+        try:
+            usage: OVODailyUsage = await self._client.get_daily_usage(
```

Comment:
This logic seems to be repeated over all entities...

Maybe consider implementing the DataUpdateCoordinator. It would reduce a lot of code, probably API call's as well, and would also be able to take care of the availability handling?

<https://developers.home-assistant.io/docs/integration_fetching_data/#coordinated-single-api-poll-for-data-for-all-entities>